plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -373,11 +382,17 @@ public void checkVersionCompatibility(List<Plugin> pluginsToBeDownloaded) {
      * Determines if there is an update center for the version of Jenkins in the war file. If so, sets jenkins update
      * center url String to include Jenkins Version. Otherwise, sets update center url to match the update center in
      * the configuration class
+     *
+     * Rules:
+     * jenkins version  | use if readable (HEAD)                    | use prior value anyway
+     *      YES         | http://update-center.jenkins.io/(version) | http://update-center.jenkins.io
+     *      NO          |                                           | http://update-center.jenkins.io
      */
     public void checkAndSetLatestUpdateCenter() {
         //check if version specific update center
         if (jenkinsVersion != null && !StringUtils.isEmpty(jenkinsVersion.toString())) {
-            String jenkinsVersionUcLatest = cfg.getJenkinsUc() + "/" + jenkinsVersion;
+            // TODO: rework to insert jenkinsVersion before the file portion of the URL
```

Comment:
is this still relevant?

Reply:
You are right. Pointing out, once again, the value of reviewing pull requests.